### DISCUSSION TITLE
Split multiple instances of the same dependency

### DISCUSSION THREAD

**Discussion by sarahhodne** [renovatebot/renovate#13537] [Request Help]
We have a large Terraform monorepo, and have a pattern that shows up fairly frequently where we have two Helm releases deploying more or less the same thing, but a "canary" release and a "production" release, something like this:

```hcl
resource "helm_release" "service_prod" {
  name = "service-prod"
  repository = "https://example.com"
  chart = "service"
  version = "1.0.0"
}

resource "helm_release" "service_canary" {
  name = "service-canary"
  repository = "https://example.com"
  chart = "service"
  version = "1.0.0"
}
```

Currently, whenever the upstream chart has a version bump, we get a PR updating both of these releases. We normally want to upgrade the canary release first, and then the production release, and we've been doing this by manually creating a new PR just bumping the canary release, and merging that, but I'm wondering if we can configure Renovate to automate this?

One of the following would work great, but any other pattern that would help automate this would be great as well.

- When the chart version bumps, Renovate files two PRs: One for the production service, and another for the canary service.
- When the chart version bumps, Renovate files one PR for the canary release. Once that's merged, Renovate files another to match the canary version.

**Comment by viceice** [comment]
If they are in different files, then you can use `packageFile` or `packageFileDir` or `parentDir` as `additionalBranchPrefix`. Search the discussions for samples.

- https://docs.renovatebot.com/templates/
- https://docs.renovatebot.com/configuration-options/#additionalbranchprefix

**Reply by rarkins** [reply]
If that doesn't satisfy, please look into the debug logs for `packageFiles with updates` and copy/paste here the entries for both the canary and non-canary packages you're referring to

**Reply by sarahhodne** [reply]
We have them in the same file, so splitting by file doesn't really work. Here's one example:

```JSON
{
  "packageFile": "terraform/aws_vpc_cni.tf",
  "deps": [
    {
      "currentValue": "1.1.12",
      "depType": "helm_release",
      "registryUrls": [
        "https://aws.github.io/eks-charts"
      ],
      "depName": "aws-vpc-cni",
      "datasource": "helm",
      "depIndex": 0,
      "updates": [],
      "warnings": [],
      "versioning": "hashicorp",
      "sourceUrl": "https://github.com/aws/amazon-vpc-cni-k8s",
      "currentVersion": "1.1.12",
      "fixedVersion": "1.1.12"
    },
    {
      "currentValue": "1.1.12",
      "depType": "helm_release",
      "registryUrls": [
        "https://aws.github.io/eks-charts"
      ],
      "depName": "aws-vpc-cni",
      "datasource": "helm",
      "depIndex": 1,
      "updates": [],
      "warnings": [],
      "versioning": "hashicorp",
      "sourceUrl": "https://github.com/aws/amazon-vpc-cni-k8s",
      "currentVersion": "1.1.12",
      "fixedVersion": "1.1.12"
    },
    {
      "skipReason": "local",
      "depIndex": 2,
      "updates": []
    }
  ]
}
```

Looks like they're identical except for the `depIndex`, and if I'm reading [the docs](https://docs.renovatebot.com/templates/) correctly that's not an available template variable?

**Reply by rarkins** [reply]
What does the raw helm look like which they were extracted from?

**Reply by sarahhodne** [reply]
@rarkins Sorry for delay, I had missed the reply, this is what the raw Terraform looks like:

```hcl
locals {
  aws_node_base_values = "long_string_with_yaml"
}

resource "helm_release" "aws_node" {
  name        = "aws-node"
  namespace   = "kube-system"
  repository  = "https://aws.github.io/eks-charts"
  chart       = "aws-vpc-cni"
  version     = "1.1.10"
  max_history = 10

  values = [
    local.aws_node_base_values,
    "<another_long_yaml_string>",
  ]

  # and a few repetitions of these
  set {
    name = "foo"
    value = "bar"
  }
}


resource "helm_release" "aws_node_canary" {
  name        = "aws-node-canary"
  namespace   = "kube-system"
  repository  = "https://aws.github.io/eks-charts"
  chart       = "aws-vpc-cni"
  version     = "1.1.10"
  max_history = 10

  values = [
    local.aws_node_base_values,
  ]

  # and a few repetitions of these
  set {
    name = "foo"
    value = "bar"
  }
}
```

We have multiple files with similar setups, I removed most of the application config (the `values` and `set` statements) that's inserted since that's wildly different between the different services that use these patterns. The only reliable difference we have between the two are the resource names (`aws_node` / `aws_node_canary`) and the release names (`aws-node` / `aws-node-canary`).

**Reply by rarkins** [reply]
It seems we use the "chart" value for depName, and not the "name" field. We presumably could switch, but I'm not sure if it would have undesirable side effects.
